google/cloud/spanner_v1/pool.py

```diff
@@ -199,21 +204,47 @@ def bind(self, database):
                 _metadata_with_leader_aware_routing(database._route_to_leader_enabled)
             )
         self._database_role = self._database_role or self._database.database_role
-        request = BatchCreateSessionsRequest(
-            database=database.name,
-            session_count=self.size - self._sessions.qsize(),
-            session_template=Session(creator_role=self.database_role),
-        )
 
-        while not self._sessions.full():
-            resp = api.batch_create_sessions(
-                request=request,
-                metadata=metadata,
+        with trace_call("CloudSpanner.BatchCreateSessions", self) as span:
+            requested_session_count = self.size - self._sessions.qsize()
+            request = BatchCreateSessionsRequest(
+                database=database.name,
+                session_count=requested_session_count,
+                session_template=Session(creator_role=self.database_role),
+            )
+
+            if requested_session_count > 0:
+                span.add_event(
```

Comment:
Clearly `span` here could be `None` if opentelemetry is not installed (see https://github.com/googleapis/python-spanner/blob/b2e67626739c05e8dc2eb81e522ff4931f9fac45/google/cloud/spanner_v1/_opentelemetry_tracing.py#L56).

In that case it can throw `AttributeError: 'NoneType' object has no attribute 'add_event'`

Reply:
@harshachinta I added a guard to it.